### PR TITLE
Group work fix on instructorGradebook

### PR DESCRIPTION
**Pull request by zhining4**
That pull request is to address the issue in #2927  and I will use this branch for later fix based on Matt's comments.

### CODE REVIEW

File: pages/instructorGradebook/instructorGradebook.sql [hunk 1]
```diff
@@ -49,7 +61,10 @@ scores AS (
         course_users AS u
         CROSS JOIN course_assessments AS a
         LEFT JOIN group_users AS gu ON (u.user_id = gu.user_id)
-        LEFT JOIN course_scores AS s ON (((s.user_id = u.user_id)OR(s.group_id = gu.group_id)) AND s.assessment_id = a.id)
+        LEFT JOIN groups AS gr ON (gr.id = gu.group_id)
+        LEFT JOIN course_scores AS s ON (((s.user_id = u.user_id) OR (s.group_id = gu.group_id)) AND s.assessment_id = a.id)
+     WHERE 
+         gr.deleted_at IS NULL
```
Comment (tdy) [inline pages/instructorGradebook/instructorGradebook.sql:67]: Just to confirm, this part is the main fix for the bug, right? And the rest of the changes are purely for reformatting?

Comment (zhining4) [reply pages/instructorGradebook/instructorGradebook.sql:67]: Yeah. That bug occurs when old groups deleted and the student join a new group, I did not check old groups' delete_at.

File: pages/instructorGradebook/instructorGradebook.sql [hunk 2]
```diff
@@ -1,26 +1,38 @@
 -- BLOCK course_assessments
 SELECT
-    a.id AS assessment_id,a.number AS assessment_number,
+    a.id AS assessment_id,
+    a.number AS assessment_number,
```
Comment (mwest1066) [inline pages/instructorGradebook/instructorGradebook.sql:4]: While I appreciate the effort to clean up code, please don't introduce additional unrelated changes (such as code reformatting) into bug fix PRs. It makes it hard for reviewers to see what's actually changed in the PR.

File: pages/instructorGradebook/instructorGradebook.sql (hunk 1 above)
Comment (mwest1066) [reply pages/instructorGradebook/instructorGradebook.sql:67]: While this is a bug, I don't think it's actually the bug here. The problems with the gradebook are showing up for courses that aren't even using groups at all. I think the problem is that you are JOINing over all `group_users`, even if they are unrelated to the current course_instance or assessment.

Comment (mwest1066) [reply pages/instructorGradebook/instructorGradebook.sql:67]: I submitted PR #2965 which I think is a fix for this bug. But please check that PR!

**Comment by mwest1066** [comment]
Closing in favor of #2965
